src/engine/strat_engine/thinpool.rs

```diff
@@ -333,25 +333,25 @@ impl ThinPool {
     /// Return the FlexDevsSave data structure for variable length metadata.
     /// May return an error if mapper can not locate the UUID corresponding
     /// to a device node.
-    pub fn flexdevssave(&self) -> EngineResult<FlexDevsSave> {
-        Ok(FlexDevsSave {
-               meta_dev: self.mdv_segments
-                   .iter()
-                   .map(|bseg| (*bseg.blockdev.borrow().uuid(), bseg.start, bseg.length))
-                   .collect::<Vec<_>>(),
-               thin_meta_dev: self.meta_segments
-                   .iter()
-                   .map(|bseg| (*bseg.blockdev.borrow().uuid(), bseg.start, bseg.length))
-                   .collect::<Vec<_>>(),
-               thin_data_dev: self.data_segments
-                   .iter()
-                   .map(|bseg| (*bseg.blockdev.borrow().uuid(), bseg.start, bseg.length))
-                   .collect::<Vec<_>>(),
-               thin_meta_dev_spare: self.meta_spare_segments
-                   .iter()
-                   .map(|bseg| (*bseg.blockdev.borrow().uuid(), bseg.start, bseg.length))
-                   .collect::<Vec<_>>(),
-           })
+    pub fn flexdevssave(&self) -> FlexDevsSave {
```

Comment:
Ah yes, this works!

We end up with 
```
            flex_devs: self.thin_pool.record(),
            thinpool_dev: self.thin_pool.record(),
```
which is a little strange, but seems to DTRT...